spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -133,59 +178,36 @@ def _handle_input_request(self, msg):
         # before entering readline mode.
         self.kernel_client.iopub_channel.flush()
         self._input_ready = True
-        self._executing = False
 
-        # While the widget thinks only one input is going on,
-        # other functions can be sending messages to the kernel.
-        # This must be properly processed to avoid dropping messages.
-        # If the kernel was not ready, the messages are queued.
-        if len(self._input_queue) > 0:
-            msg = self._input_queue[0]
-            del self._input_queue[0]
-            self.kernel_client.input(msg)
-            return
+        prompt, password = msg['content']['prompt'], msg['content']['password']
 
-        def callback(line):
-            if not self.is_debugging():
-                return self.kernel_client.input(line)
-            line = line.strip()
+        # Check if this is a duplicate that we shouldn't reprint.
+        # This can happen when sending commands to pdb from the frontend.
+        print_prompt = (not self._reading
+                        or not (prompt, password) == self._previous_prompt)
 
-            # Save history to browse it later
-            self._pdb_line_num += 1
-            self.add_to_pdb_history(self._pdb_line_num, line)
+        self._previous_prompt = (prompt, password)
 
-            if line:
-                self._last_pdb_cmd = line
+        if print_prompt:
+            self._highlighter.highlighting_on = True
```

Comment:
Should the highlighter only be turned on for a pdb request? Here it is turned on for any request.

Reply:
Yes, please only turn it on for Pdb requests.